### PR TITLE
Buybutton

### PR DESCRIPTION
**Pull request by ak2253**

### CODE REVIEW

File: tests/buy_button_test.py [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+KEY_MESSAGE = "message"
+KEY_ERROR = "error"
+KEY_ID = "id"
+KEY_QUANTITY = "quantity"
+KEY_EXPECTED = "expected"
+KEY_PARAMS = "parameters"
+KEY_USER_ID = "user_id"
+KEY_AMOUNT = "AMOUNT"
+
+class BuyButtonTest(DatabaseTest):
```
Comment (dvkm) [inline tests/buy_button_test.py]: Purchase test? Buy test? We aren't exactly testing the button itself.

File: tests/buy_button_test.py [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+    def setUp(self):
+        super().setUp()
+        self.user1_id = 1
+        self.user2_id = 2
+        self.user3_id = 3
+        trans1 = Transaction (
+            user_id=self.user1_id,
+            ticket_amount=1500,
+            activity="Transaction Test",
```
Comment (dvkm) [inline tests/buy_button_test.py]: Not a transaction test. Maybe buy test?

File: tests/buy_button_test.py [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+            result = json.loads(res.data.decode("utf-8"))
+            self.assertDictEqual(test_case[KEY_EXPECTED], result)
+            self.assertEqual(
+                test_case[KEY_AMOUNT],
+                db.session.query(func.sum(Transaction.ticket_amount))
+                .filter(Transaction.user_id == test_case[KEY_USER_ID])
+                .scalar()
+            )
+
+    def test_buy_button_success(self):
```
Comment (dvkm) [inline tests/buy_button_test.py]: here too. We aren't really testing buy button. Applies to all other method names.

**Review by dvkm — CHANGES_REQUESTED** [review]

File: tests/buy_button_test.py (hunk 2 above)
Comment (ak2253) [reply tests/buy_button_test.py]: Its still a transaction so I'll keep as "Transaction Test"

**Review by speshori — APPROVED** [review]

**Review by dvkm — APPROVED** [review]

**Pull request merged**